lib/rhc/ssh_helpers.rb

```diff
@@ -256,29 +257,43 @@ def save_snapshot(app, filename, for_deployment=false, ssh_executable=nil)
       ssh_stderr = " 2>/dev/null"
       debug ssh_cmd
 
+      # Default timeout is 30 minutes, more than enough time to ensure the snapshot is copied successfully.
+      options.timeout ? exec_timeout = options.timeout : exec_timeout = 1800
+
       say "Pulling down a snapshot of application '#{app.name}' to #{filename} ... "
 
       begin
         if !RHC::Helpers.windows?
+          status = 1
+          output = ""
+          Timeout::timeout(exec_timeout) {
             status, output = exec(ssh_cmd + (debug? ? '' : ssh_stderr))
```

Comment:
Hm, why don't we conditionally set `ssh_stderr = debug? '' : ' 2>/dev/null'` above and then unconditionally interpolate `exec(ssh_cmd + ssh_stderr)` here? But I guess that's outside of scope for this PR.